src/common/config_proxy.h

```diff
@@ -124,11 +119,16 @@ class ConfigProxy {
   ConfigValues* operator->() noexcept {
     return &values;
   }
-#ifdef WITH_SEASTAR
+  ConfigValues get_config_values() const {
+    std::lock_guard l{lock};
+    return values;
+  }
   void set_config_values(const ConfigValues& val) {
+#ifndef WITH_SEASTAR
```

Comment:
@idryomov  I had presumed that the difference was likely related to this change to pre-processor directives in this common code. Being fairly new to the project, I lack depth of knowledge here, but I was under the impression that the test failure on Octopus might be explained by the fact that WITH_SEASTAR was defined on master but not Octopus. Let me know if that might make sense?

Reply:
I don't think `WITH_SEASTAR` is involved here.  Let's see what @cbodley says but in the meantime it would great if you could recall exactly what locking appeared to be racy and how the failure looked like.

Reply:
@idryomov this assertion was failing: https://github.com/ceph/ceph/pull/41763/files#diff-7e4322dc5c1f46fd59b989da9d0940833d5d1533b1cfab5d5ac6bc455298e2aaR430 . I believe that I described it as "racy" because I saw the assertion pass on 1 out 4 test runs (and assumed it was related to this lock that is only acquired when WITH_SEASTAR is not defined).

Reply:
OK, so this bug exists in master:

https://tracker.ceph.com/issues/42788
https://github.com/ceph/ceph/pull/42237

Please drop the workaround.

Reply:
Thanks for helping chase that down @idryomov ! I've updated the PR to remove that workaround.